### PR TITLE
Issue 1542/refactor tags hook

### PR DESCRIPTION
**Pull request by kaulfield23**
## Description
This PR refactors the `tagModel` into `useTag` hook.

## Changes

* Create `useTag` and fix its relevant codes
* Create `useTagMutation` and fix its relevant codes
* Delete `tagModel.ts`

## Notes to reviewer
Didn't delete `tagRepo` since `assignTagToPerson` and `removeTagFromPerson` are used in `ViewDataModel.ts`

Because I noticed that `assignTagToPerson` and `removeTagFromPerson` are used in other places, I decided to make a `useTagMutation` hook instead of put those methods in `useTag`. 
This will be useful in the future, once `tagRepo` is removed, as you'll be able to use `useTagMutation` to other hooks.

## Related issues
part of issue-1542

### CODE REVIEW

**Review by rebecarubio — COMMENTED** [review]
Nice fix! This lgtm but I would like @richardolsson to double check, as we are getting familiar with this new pattern :)

File: src/features/tags/hooks/useTag.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import { ZetkinTag } from 'utils/types/zetkin';
+import { tagAssigned, tagLoad, tagLoaded, tagUnassigned } from '../store';
+import { useApiClient, useEnv } from 'core/hooks';
+
+interface UseTagReturn {
+  assignToPerson: (personId: number, value?: string) => void;
+  removeFromPerson: (personId: number) => Promise<void>;
+  tagData: ZetkinTag | null;
+  tagError: unknown | null;
+  tagIsLoading: boolean;
```
Comment (richardolsson) [inline src/features/tags/hooks/useTag.ts:14]: The name of the hook already tells us that the hook will return a tag, so there is no need for these prefixes. I would call these properties:

```suggestion
  data: ZetkinTag | null;
  error: unknown | null;
  isLoading: boolean;
```

File: src/features/tags/hooks/useTag.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+  const assignToPerson = async (personId: number, value?: string) => {
+    const data = value ? { value } : undefined;
+    const tag = await apiClient.put<ZetkinTag>(
+      `/api/orgs/${orgId}/people/${personId}/tags/${tagId}`,
+      data
+    );
+    store.dispatch(tagAssigned([personId, tag]));
+  };
+  const getTag = () => {
```
Comment (richardolsson) [inline src/features/tags/hooks/useTag.ts:30]: This function is unnecessary. Inline it's contents inside the hook instead. You can just grab the return value from `loadItemIfNecessary` and use that further down the function where you are currently calling `getTag()`.

File: src/features/tags/hooks/useTag.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+import { loadItemIfNecessary } from 'core/caching/cacheUtils';
+import { RootState } from 'core/store';
+import { ZetkinTag } from 'utils/types/zetkin';
+import { tagAssigned, tagLoad, tagLoaded, tagUnassigned } from '../store';
+import { useApiClient, useEnv } from 'core/hooks';
+
+interface UseTagReturn {
+  assignToPerson: (personId: number, value?: string) => void;
+  removeFromPerson: (personId: number) => Promise<void>;
```
Comment (richardolsson) [inline src/features/tags/hooks/useTag.ts:11]: I like how these mutation functions are returned.

**Review by richardolsson — CHANGES_REQUESTED** [review]
Nice work in general! There are some changes that I think would make the code cleaner though. I've commented below.

**Review by rebecarubio — APPROVED** [review]
lgtm 🚀

**Pull request merged**